compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -11905,12 +11920,12 @@ OMR::Z::TreeEvaluator::arraycmplenEvaluator(TR::Node * node, TR::CodeGenerator *
    cg->stopUsingRegister(secondLen);
    cg->stopUsingRegister(orgLen);
 
-
    cg->decReferenceCount(elemsExpr);
    cg->decReferenceCount(firstBaseAddr);
    cg->decReferenceCount(secondBaseAddr);
    cursor->setDependencyConditions(dependencies);
 
+   generateS390LabelInstruction(cg, TR::InstOpCode::label, node, shortCircuitLabel);
```

Comment:
Don't we have an internal control flow here that ends at shortCircuitLabel here?

Reply:
I thought the control flow is when there is a loop involved which is not the case in this method. Was I wrong?

Reply:
No, that is not true. Here you have branch in your code that jumps to done label and unsuccessful branch continues execution. Local RA without properly defined internal control flow may spill register in one path but not other.